src/common/translate_item.js

```diff
@@ -93,7 +93,62 @@ Zotero.Translate.ItemSaver.prototype = {
 	 */
 	saveItems: async function (items, attachmentCallback, itemsDoneCallback=()=>0) {
 		items = await this._processItems(items);
-	
+
+		// For saving webpages, attach SingleFile page data to attachments
+		let pageContent;
+
+		for (let x = 0; x < items.length; x++) {
+			if (items[x].itemType !== 'webpage') {
+				continue;
+			}
+
+			for (let y = 0; y < items[x].attachments.length; y++) {
+				if (items[x].attachments[y].mimeType !== 'text/html') {
+					continue;
+				}
+
+				// If we have not done so already, inject SingleFile and retrieve
+				// the page content
+				if (!pageContent) {
```

Comment:
This was sort of old logic. I was going off of a multiple translator that saved a snapshot for every multiple item. This prevented us from redoing the SingleFile retrieval every time. However, that was bad logic on the part of the translator so it's no longer particularly relevant, but just in case, it speeds things up dramatically.

Also, the SingleFile logic has mostly been extracted out now.